src/Draco.Compiler/Internal/Documentation/DocumentationElement.cs

```diff
@@ -0,0 +1,102 @@
+using System.Linq;
+using System.Xml.Linq;
+using Draco.Compiler.Internal.Symbols;
+
+namespace Draco.Compiler.Internal.Documentation;
+
+/// <summary>
+/// Represents single documentation element.
+/// </summary>
+internal abstract record class DocumentationElement
+{
+    /// <summary>
+    /// Creates a markdown representation of this documentation element.
+    /// </summary>
+    /// <returns>The documentation in markdown format.</returns>
+    public abstract string ToMarkdown();
+
+    /// <summary>
+    /// Creates an XML representation of this documentation element.
+    /// </summary>
+    /// <returns>The documentation in XML format.</returns>
+    public abstract XNode ToXml();
+}
+
+/// <summary>
+/// Represents regular text inside documentation.
+/// </summary>
+/// <param name="Text">The text represented by this element.</param>
+internal sealed record class TextDocumentationElement(string Text) : DocumentationElement
+{
+    public override string ToMarkdown() => this.Text;
+
+    public override XText ToXml() => new XText(this.Text);
+}
+
+/// <summary>
+/// A single parameter.
+/// </summary>
+/// <param name="ParameterLink">The link to given parameter.</param>
+/// <param name="Elements">The <see cref="DocumentationElement"/>s that are contained in the description of this parameter.</param>
+internal sealed record class ParameterDocumentationElement(SymbolReferenceDocumentationElement ParameterLink, ImmutableArray<DocumentationElement> Elements) : DocumentationElement
+{
+    public override string ToMarkdown() => $"- {this.ParameterLink.ToMarkdown()}: {string.Join("", this.Elements.Select(x => x.ToMarkdown()))}";
+
+    public override XElement ToXml() => new XElement("param",
+        new XAttribute("name", this.ParameterLink.DisplayText),
+        this.Elements.Select(x => x.ToXml()));
+}
+
+/// <summary>
+/// A single type parameter.
+/// </summary>
+/// <param name="ParameterLink">The link to given type parameter.</param>
+/// <param name="Elements">The <see cref="DocumentationElement"/>s that are contained in the description of this type parameter.</param>
+internal sealed record class TypeParameterDocumentationElement(SymbolReferenceDocumentationElement ParameterLink, ImmutableArray<DocumentationElement> Elements) : DocumentationElement
+{
+    public override string ToMarkdown() => $"- {this.ParameterLink.ToMarkdown()}: {string.Join("", this.Elements.Select(x => x.ToMarkdown()))}";
+
+    public override XElement ToXml() => new XElement("typeparam",
+        new XAttribute("name", this.ParameterLink.DisplayText),
+        this.Elements.Select(x => x.ToXml()));
+}
+
+/// <summary>
+/// A link to some symbol in code.
+/// </summary>
+/// <param name="ReferencedSymbol">The symbol that is linked.</param>
+/// <param name="DisplayText">The text that should be displayed in the link.</param>
+internal record class SymbolReferenceDocumentationElement(Symbol? ReferencedSymbol, string DisplayText) : DocumentationElement
```

Comment:
Let's rename this to something more terse, since symbolic refs are likely the only thing we need, I'd suggest `ReferenceDocumentationElement`